src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs

```diff
@@ -29,19 +28,12 @@ internal abstract partial class AbstractEditorFactory : IVsEditorFactory, IVsEdi
         private readonly Package _package;
         private readonly IComponentModel _componentModel;
         private Microsoft.VisualStudio.OLE.Interop.IServiceProvider _oleServiceProvider;
-        private readonly IVsEditorAdaptersFactoryService _editorAdaptersFactoryService;
-        private readonly IContentTypeRegistryService _contentTypeRegistryService;
-        private readonly IWaitIndicator _waitIndicator;
         private bool _encoding;
 
-        protected AbstractEditorFactory(Package package)
+        protected AbstractEditorFactory(Package package, IComponentModel componentModel)
         {
             _package = package ?? throw new ArgumentNullException(nameof(package));
```

Comment:
It looks like we can probably eliminate the package being passed entirely. The only use I'm seeing (assuming I'm not missing something) is we provide it as the ServiceProvider which is used in just one place. But we might also be able to use the _oleServiceProvider too at that point since this factory has already been sited?